src/frequenz/sdk/microgrid/_data_pipeline.py

```diff
@@ -0,0 +1,224 @@
+# License: MIT
+# Copyright © 2023 Frequenz Energy-as-a-Service GmbH
+
+"""Create, connect and own instances of data pipeline components.
+
+Provides SDK users direct access to higher level components of the data pipeline,
+eliminating the boiler plate code required to setup the DataSourcingActor and the
+ResamplingActor.
+"""
+
+from __future__ import annotations
+
+import typing
+from dataclasses import dataclass
+
+from frequenz.channels import Broadcast, Sender
+
+# A number of imports had to be done inside functions where they are used, to break
+# import cycles.
+#
+# pylint: disable=import-outside-toplevel
+if typing.TYPE_CHECKING:
+    from ..actor import (
+        ComponentMetricRequest,
+        ComponentMetricsResamplingActor,
+        DataSourcingActor,
+        ResamplerConfig,
+    )
+    from ..timeseries.ev_charger_pool import EVChargerPool
+    from ..timeseries.logical_meter import LogicalMeter
+
+
+@dataclass
+class _ActorInfo:
+    """Holds instances of core data pipeline actors and their request channels."""
+
+    actor: "DataSourcingActor | ComponentMetricsResamplingActor"
+    channel: Broadcast["ComponentMetricRequest"]
+
+
+class _DataPipeline:
+    """Create, connect and own instances of data pipeline components.
+
+    Provides SDK users direct access to higher level components of the data pipeline,
+    eliminating the boiler plate code required to setup the DataSourcingActor and the
+    ResamplingActor.
+    """
+
+    def __init__(
+        self,
+        resampler_config: ResamplerConfig,
+    ) -> None:
+        """Create a `DataPipeline` instance.
+
+        Args:
+            resampler_config: Config to pass on to the resampler.
+        """
+        from ..actor import ChannelRegistry
+
+        self._resampler_config = resampler_config
+
+        self._channel_registry = ChannelRegistry(name="Data Pipeline Registry")
+
+        self._data_sourcing_actor: _ActorInfo | None = None
+        self._resampling_actor: _ActorInfo | None = None
+        self._logical_meter: "LogicalMeter" | None = None
+        self._ev_charger_pools: dict[frozenset[int], "EVChargerPool"] = {}
+
+    def logical_meter(self) -> LogicalMeter:
```

Comment:
I wonder if this be suitable to make it a @property

Reply:
I don't know, in my opinion, this does actual work, so it is not a property.  Also, it is internal and not the public api, so it doesn't matter.

Reply:
I was mainly considering to use this kind of syntax `microgrid.logical_meter.grid_power()` instead but as you have said it doesn't matter as it is internal